### ISSUE TITLE
Efficient way to save histories

### ISSUE THREAD

**Issue by zsunberg** [JuliaPOMDP/POMDPToolbox.jl#13]
Right now, JLD takes much to long to save simulation histories. It takes me 6 minutes to run 1200 simulations even with an online MCTS solver running, but 30 minutes and 700MB later, JLD still has not saved the histories (I am trying to save them as a vector of HistoryRecorder objects). Does anyone have any ideas about how to store these better, or do I just need to avoid storing histories altogether?

**Comment by tawheeler** [comment]
Are your types immutable? That makes a huge difference. JLD has a hard time with abstract arrays and types with pointers.

**Comment by zsunberg** [comment]
Yeah, our type system is not great for serialization and structured storage  - even though the states are immutable, they contain pointers to other objects. I think that is the right way to do what we are doing in the simulation code, but yeah, obviously that's hard for HDF5 to deal with. So maybe this problem is specific to the domain I am working in.

**Comment by zsunberg** [comment]
I guess maybe using vec() on the states and then saving them as a large matrix might be the simplest way
